src/interpolation/details/ArborX_InterpDetailsMovingLeastSquaresCoefficients.hpp

```diff
@@ -69,167 +264,24 @@ movingLeastSquaresCoefficients(ExecutionSpace const &space,
-      Kokkos::MDRangePolicy<ExecutionSpace, Kokkos::Rank<2>>(
-          space, {0, 0}, {num_targets, num_neighbors}),
-      KOKKOS_LAMBDA(int const i, int const j) {
-        auto basis = evaluatePolynomialBasis<degree>(source_ref_target(i, j));
-        for (int k = 0; k < poly_size; k++)
-          p(i, j, k) = basis[k];
-      });
-
-  Kokkos::Profiling::popRegion();
-  Kokkos::Profiling::pushRegion(
-      "ArborX::MovingLeastSquaresCoefficients::moment");
-
-  // We then create what is called the moment matrix, which is A = P^T.PHI.P. By
-  // construction, A is symmetric.
-  Kokkos::View<CoefficientsType ***, MemorySpace> a(
-      Kokkos::view_alloc(space, Kokkos::WithoutInitializing,
-                         "ArborX::MovingLeastSquaresCoefficients::moment"),
-      num_targets, poly_size, poly_size);
-  Kokkos::parallel_for(
-      "ArborX::MovingLeastSquaresCoefficients::moment_computation",
-      Kokkos::MDRangePolicy<ExecutionSpace, Kokkos::Rank<3>>(
-          space, {0, 0, 0}, {num_targets, poly_size, poly_size}),
-      KOKKOS_LAMBDA(int const i, int const j, int const k) {
-        CoefficientsType tmp = 0;
-        for (int l = 0; l < num_neighbors; l++)
-          tmp += p(i, l, j) * p(i, l, k) * phi(i, l);
-        a(i, j, k) = tmp;
-      });
-
-  Kokkos::Profiling::popRegion();
-  Kokkos::Profiling::pushRegion(
-      "ArborX::MovingLeastSquaresCoefficients::pseudo_inverse_svd");
-
-  // We need the inverse of A = P^T.PHI.P, and because A is symmetric, we can
-  // use the symmetric SVD algorithm to get it.
-  symmetricPseudoInverseSVD(space, a);
-  // Now, A = [P^T.PHI.P]^-1
-
-  Kokkos::Profiling::popRegion();
-  Kokkos::Profiling::pushRegion(
-      "ArborX::MovingLeastSquaresCoefficients::coefficients_computation");
-
-  // Finally, the result is produced by computing p(0).A.P^T.PHI
-  Kokkos::View<CoefficientsType **, MemorySpace> coeffs(
+  Kokkos::View<CoefficientsType **, MemorySpace> coefficients(
       Kokkos::view_alloc(
           space, Kokkos::WithoutInitializing,
           "ArborX::MovingLeastSquaresCoefficients::coefficients"),
-      num_targets, num_neighbors);
-  Kokkos::parallel_for(
-      "ArborX::MovingLeastSquaresCoefficients::coefficients_computation",
-      Kokkos::MDRangePolicy<ExecutionSpace, Kokkos::Rank<2>>(
-          space, {0, 0}, {num_targets, num_neighbors}),
-      KOKKOS_LAMBDA(int const i, int const j) {
-        CoefficientsType tmp = 0;
-        for (int k = 0; k < poly_size; k++)
-          tmp += a(i, 0, k) * p(i, j, k) * phi(i, j);
-        coeffs(i, j) = tmp;
-      });
-
-  Kokkos::Profiling::popRegion();
-  return coeffs;
+      source_points.extent_int(0), source_points.extent_int(1));
+
+  MovingLeastSquaresCoefficientsKernel<CRBFunc, PolynomialDegree, SourcePoints,
+                                       TargetAccess, decltype(coefficients),
+                                       ExecutionSpace>
+      kernel(space, target_access, source_points, coefficients);
```

Comment:
Can we reorder the template arguments so we don't need to specify all of them here?

Reply:
I'll put the CRBF and PolynomialDegree at the end and set defaults. The 4 others must be determined by the outside.